### DISCUSSION TITLE
How to manage expired connections in AsyncConnectionPool?

### DISCUSSION THREAD

**Discussion by tscu-vid** [psycopg/psycopg#891] [Q&A]
Using psycopg_pool 3.2.2

Hello, I have an async connection pool which use for a FastAPI application. 
I create the pool at startup, and continue to use it sharing the pool among the different endpoints. 
However, when the `max_lifetime` is reached, the connections are closed and the timeout exception is raised. This even if I constantly (before each `async with conn.connection() as async_conn:` call) call pool.check() (which from psycopg_pool 3.1.8 shall check if any of the connection is expired, and if so, replace with a new one).
I see that the `max_lifetime` check is performed, the connection is closed if too old, the `run_task(AddConnection(self))` executed. However, the real creation of the new connection is never performed (e.g. the `_run` method of the `AddConnection` object).
Am I doing anything wrong in the usage of the library?

**Comment by dvarrazzo** [comment]
Because you don't share anything, for instance the way you create a pool, it's difficult to know if you are doing something wrong.

However, likely, you are doing something wrong, because connection re-creation is well tested.

You can increase the logging level of the pool to check from the message if something unexpected is happening. You can also reduce the connection timeout to get clearer errors if connection fails for some reason. Both these procedures are explained in the pool documentation page.
